src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+    { 12, 44, 0, { 0, 0, 0 } },
+};
+
+// Collision Check data
+// sColChkInfoInit
+CollisionCheckInfoInit D_80A9C234 = { 0, 0xC, 0x1E, 0xFF };
+
+Vec3f D_80A9C23C[] = { { 0.0, 0.707099974155, 0.707099974155 },
+                       { 0.707099974155, 0.707099974155, 0.0 },
+                       { 0.0, 0.707099974155, -0.707099974155 },
+                       { -0.707099974155, 0.707099974155, 0.0 } };
+
+s16 D_80A9C26C[] = { 0x006C, 0x0066, 0x0060, 0x0054, 0x0042, 0x0037, 0x002A, 0x0026 };
+
+// sInitChain
+InitChainEntry D_80A9C27C[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 400, ICHAIN_CONTINUE),         ICHAIN_F32_DIV1000(gravity, -3200, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(minVelocityY, -17000, ICHAIN_CONTINUE), ICHAIN_F32(uncullZoneForward, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 100, ICHAIN_CONTINUE),         ICHAIN_F32(uncullZoneDownward, 120, ICHAIN_STOP),
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9AFA0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9AFAC.s")
+s32 D_80A9C294[] = { 0x0500B9D0, 0x06000140, 0x06000140 };
+
+extern Gfx D_060002E0[];
+extern Gfx D_040355E0[]; // bush fragments 1
+extern Gfx D_040356A0[]; // bush fragments 2
+
+void func_80A9AFA0(EnKusa* this, EnKusaActionFunc actionFunc) {
+    this->unk_19C = 0;
+    this->actionFunc = actionFunc;
+}
+
+// Snap to floor
+s32 func_80A9AFAC(EnKusa* this, GlobalContext* globalCtx, f32 arg2) {
+    CollisionPoly* sp34;
+    f32 sp28;
+    Vec3f sp30;
+    UNK_TYPE sp24;
+    f32 temp_f0;
+
+    sp30.x = this->actor.posRot.pos.x;
+    sp30.y = this->actor.posRot.pos.y + 30.0f;
+    sp30.z = this->actor.posRot.pos.z;
+    temp_f0 = func_8003C9A4(&globalCtx->colCtx, &sp28, &sp24, &this->actor, &sp30);
+    if (-32000.0f < temp_f0) {
+        this->actor.posRot.pos.y = temp_f0 + arg2;
+        Math_Vec3f_Copy(&this->actor.initPosRot, &this->actor.posRot);
+        return true;
+    }
+    osSyncPrintf(VT_COL(YELLOW, BLACK));
+    // Translation: Failure attaching to ground
+    osSyncPrintf("地面に付着失敗(%s %d)\n", "../z_en_kusa.c", 0x143);
+    osSyncPrintf(VT_RST);
+    return false;
+}
+
+// Spawn Collectible
+void func_80A9B07C(EnKusa* this, GlobalContext* globalCtx) {
+    s16 dropParams;
+    switch (this->actor.params & 3) {
+        case 0:
+        case 2:
+            dropParams = (this->actor.params >> 8) & 0xF;
+            if (dropParams >= 0xD) {
+                dropParams = 0;
+            }
+            Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.posRot.pos, dropParams << 4);
+            break;
+
+        case 1:
+            if (Math_Rand_ZeroOne() < 0.5f) {
+                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, 0x10);
+
+            } else {
+                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, 3);
+                break;
+            }
```

Comment:
```suggestion
            if (Math_Rand_ZeroOne() < 0.5f) {
                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, ITEM00_SEEDS);
            } else {
                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, ITEM00_HEART);
            }
            break;
```
`break` should go outside the else block. Also we have enums for the collectible drop parameters.